fed/mosaic-cell/src/main/java/org/eclipse/mosaic/fed/cell/utility/CapacityUtility.java

```diff
@@ -133,8 +138,28 @@ public static void consumeCapacity(TransmissionMode mode, CNetworkProperties reg
      * @param msg V2X message.
      * @return The length of the V2X message.
      */
-    public static long getMessageLength(V2xMessage msg) {
-        return msg.getPayload().getEffectiveLength() * DATA.BYTE;
+    public static long getMessageLengthWithHeaders(V2xMessage msg, String senderOrReceiver) {
+        final CCell.CHeaderLengths headerLengths = ConfigurationData.INSTANCE.getCellConfig().headerLengths;
+        final long linkLayerHeader;
+        if (senderOrReceiver != null && (isServer(senderOrReceiver) || isTmc(senderOrReceiver))) {
+            // let's assume everything is connected via cellular link, except servers and tmcs which are connected with the backbone
+            linkLayerHeader = headerLengths.ethernetHeader;
+        } else {
+            linkLayerHeader = headerLengths.cellularHeader;
+        }
+        switch (msg.getRouting().getDestination().getProtocolType()) {
```

Comment:
> However, if users may need more freedom in definition of their headers,

They do have complete control over the header sizes already, since they can override all the proposed values in the config. I included the header sizes of the PHY layer to `ethernetHeader`/`cellularHeader` since they differ between both technologies. Moving PHY sizes to a more general `additionalOverhead` field would use the same header size for both cellular/link-based transmission. I see your point, that PHY depends on the infrastructure being used (optical vs. wires), but it is already configurable by just using different values here.

tl;dr, We could arguably just ignore PHY completely if you wish to. Still, I cannot see any value by adding the "additionalOverhead" field since the user can already configure additional header sizes (with more control) by re-configuring/overriding the proposed fields.